Miniprojectasher

asherbolotski opened the pull request:
through bore encoder to talon (works!)

asherbolotski committed:
initial version

asherbolotski committed:
initial version of project

4682cybears committed:
works!

asherbolotski committed:
Remove example command and subsystem files

Deleted Autos.java, ExampleCommand.java, and ExampleSubsystem.java from the project. These files were sample implementations and are no longer needed.

asherbolotski committed:
Refactor motor subsystem and command documentation

Added and improved Javadoc comments for MotorSubsystem and DefaultMotorCommand to clarify their responsibilities and usage. Removed unused code and simplified RobotContainer, setting the default motor target angle to 0 and returning a no-op autonomous command. Updated configuration to use constants for sensor discontinuity point.

josieammer commented on camp/asher-miniproject/src/main/deploy/example.txt:
```diff
@@ -0,0 +1,3 @@
+Files placed in this directory will be deployed to the RoboRIO into the
```
remove this unneeded file.

asherbolotski replied:
removed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,35 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
when we edit a file, we replace the WPILIB copyright with the BBHS header.

asherbolotski replied:
done

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,35 @@
+  public static final SensorDirectionValue shooterAngleSensorDirection = SensorDirectionValue.CounterClockwise_Positive;
+  public static final double encoderAbsoluteOffsetDegrees = 0;
+
+  public static class OperatorConstants {
```
remove unused code

asherbolotski replied:
no unused code

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,35 @@
+  public static final double motorSupplyVoltageTimeConstant = Constants.ctreSupplyVoltageTimeConstant;
+  public static final double ctreSupplyCurrentMaximumAmps = 50.0;
+  public static final double ctreSupplyVoltageTimeConstant = 0.02;
+  public static final double joystickDeadband = 0.1;
```
this seems to be unused. If so, remove.

asherbolotski replied:
removed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,31 @@
+
```
Need BBHS header

asherbolotski replied:
added

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/commands/DefaultMotorCommand.java:
```diff
@@ -0,0 +1,61 @@
+  private final DoubleSupplier encoderSupplier;
+  private double targetAngle;
+
+  private final double encoderDeadband = 2.0;
```
leave a comment with Units. Even better is to name this encodeDeadbandDegrees.

asherbolotski replied:
added

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/commands/DefaultMotorCommand.java:
```diff
@@ -0,0 +1,61 @@
+   */
+  @Override
+  public void execute() {
+    // Implementation details
```
Where's the implementation?!?!?

asherbolotski replied:
added :P

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+ */
+public class MotorSubsystem extends SubsystemBase{
+    private final TalonFX motor;
+        private final DutyCycleOut eeDutyCycle = new DutyCycleOut(0.0);
```
fix indentation

asherbolotski replied:
fixed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+     * @param motorSpeed The desired speed of the motor. Values are clamped between -1 (full reverse) and 1 (full forward).
+     */
+    public void setMotorSpeed(double motorSpeed){
+        eeDutyCycle.withOutput(clamp(motorSpeed, -1, 1));
```
we want to now use edu.wpi.first.math.MathUtil.clamp instead of writing our own.

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+     */
+    public static double clamp(double value, double min, double max) {
+        if (min > max) {
+            throw new IllegalArgumentException("min must not be greater than max");
```
We *never* throw an error in robotics code! We need to find a way to handle the error gracefully. You can print a warning and then do something intelligent. We don't want our RoboRio to crash in the middle of a competition because of a minor bug.

asherbolotski replied:
fixed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+    }
+
+    private double wrapDegrees(double degrees){
+        return (degrees % 360 + 540) % 360 - 180;
```
please leave a comment explaining the behavior of this code. Where is the wrap point? 
Also seems like the double modulo 360 is redundant. Maybe: (degrees + 540) % 360 - 180

asherbolotski replied:
changed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+package frc.robot.subsystems;
```
need BBHS header

asherbolotski replied:
added

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+     *
+     * @param motorSpeed The desired speed of the motor. Values are clamped between -1 (full reverse) and 1 (full forward).
+     */
+    public void setMotorSpeed(double motorSpeed){
```
reorder methods to be:
* constructor
 * public methods in alphabetical order. 
* private methods in alphabetical order.

asherbolotski replied:
reordered

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,147 @@
+        // Config CanCoder
+        CANcoderConfiguration encoderConfigs = new CANcoderConfiguration();
+        encoderConfigs.MagnetSensor.MagnetOffset = degreesToRotations(Constants.encoderAbsoluteOffsetDegrees);
+        encoderConfigs.MagnetSensor.AbsoluteSensorDiscontinuityPoint = Constants.absoluteSensorDiscontinuityPoint;
```
Constants. absoluteSensorDiscontinuityPoint does not exist. The code does not compile.

asherbolotski replied:
nuh uh

josieammer reviewed (changes requested):
The code doesn't compile. You seem to be missing the execute function in your DefaultMotorCommand and at least one variable in Constants.java. 
In the mean time, I've tried to provide comments on as much of the other code as possible. 
Also, please copy in the .gitignore from Reefscape2025/theRobot/ into your camp project.

asherbolotski committed:
Update headers and add constant for sensor discontinuity

Added project-specific headers to Constants.java and RobotContainer.java for Bishop Blanchet Robotics. Introduced absoluteSensorDiscontinuityPoint constant in Constants.java. Removed example.txt from deploy directory.

asherbolotski committed:
Remove unused joystick and operator constants

Deleted the joystickDeadband constant and the OperatorConstants inner class from Constants.java as they are no longer used. Added a .gitignore file to exclude build artifacts and IDE-specific files from version control.

asherbolotski committed:
Implement motor control logic in DefaultMotorCommand

Added logic to set motor speed based on the difference between target and current angle, with a deadband check. Also updated the end method to set the motor speed when the command finishes.

asherbolotski committed:
Refactor motor speed handling and remove custom clamp

Updated DefaultMotorCommand to stop the motor on end instead of restoring speed. MotorSubsystem now uses WPILib's clamp and stops the motor when speed is zero, removing the custom clamp implementation for cleaner code.

asherbolotski committed:
Update team number and add header to MotorSubsystem

Set the team number to 4682 in build.gradle for deployment configuration. Added a project header and ASCII art to MotorSubsystem.java for documentation and branding.

asherbolotski committed:
Update MotorSubsystem.java

asherbolotski committed:
Refactor clamp usage and remove custom implementation

Replaced the custom clamp method with MathUtil.clamp for motor speed limiting. Removed the redundant clamp method and fixed the wrapDegrees calculation for improved accuracy.

asherbolotski committed:
Refactor MotorSubsystem method order and docs

Swapped the order of getEndodgerPositionDegrees and setMotorSpeed methods for improved logical grouping. Updated method documentation to better describe functionality and usage. Reformatted code for consistency and readability.

asherbolotski committed:
Update MotorSubsystem.java

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/commands/DefaultMotorCommand.java:
```diff
@@ -0,0 +1,71 @@
+    if (Math.abs(angleDifference) <= encoderDeadbandDegrees) {
+      motorSubsystem.setMotorSpeed(0);
+    } else {
+      double motorSpeed = angleDifference > 0 ? 0.5 : -0.5;
```
Magic number. The speed of 0.5 should be declared at the top of the class as an instance variable, and then you can use that variable here.

asherbolotski replied:
changed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,129 @@
+            eeDutyCycle.withOutput(MathUtil.clamp(motorSpeed, -1, 1));
+            motor.setControl(eeDutyCycle);
+        }
+
```
Missing closing } to close out setMotorSpeed method.

asherbolotski replied:
fixed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,129 @@
+    }
+
+    /**
+     * Retrieves the position of the endodger in degrees.
```
what's an endodger? Maybe you mean encoder?

asherbolotski replied:
typo fixed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,129 @@
+     * <p>
+     * This method calculates the position of the endodger by converting the
+     * encoder's position value (in rotations) to degrees and wrapping it
+     * within a valid range of degrees (e.g., 0 to 360).
```
Fix this comment. It looks like your wrap degrees function wraps between [-180 .. 180].

asherbolotski replied:
fixed

josieammer commented on camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,129 @@
+     * within a valid range of degrees (e.g., 0 to 360).
+     * </p>
+     *
+     * @return The position of the endodger in degrees, wrapped within a valid range.
```
endodger?

asherbolotski replied:
fixed typo

josieammer reviewed (changes requested):
close. Just a few more nits to clean up.

asherbolotski committed:
pr changed

josieammer reviewed (approved):
LGTM!
